### ISSUE TITLE
Add a rule forbidding invalid syntaxes for CSS custom properties

### ISSUE THREAD

**Issue by stof** [stylelint/stylelint#8745]
### What is the problem you're trying to solve?

In my project, this kind of code was added at some point:

```css
@property --my-prop {
  syntax: "<decimal>";
}
```

This is broken because `<decimal>` is not a valid type in CSS Values. It should have been `<number>` instead.

### What solution would you like to see?

I'd like to have a stylelint rule validating strings provided for the `syntax` descriptor of `@property` to enforce that they are valid syntaxes (either as a new rule or as part of `at-rule-descriptor-value-no-unknown` where it could fit).

**Comment by jeddy3** [comment]
@stof Thanks for the request.

Good news, we already have the [`syntax-string-no-invalid`](https://stylelint.io/user-guide/rules/syntax-string-no-invalid/) rule that checks this.

It's turned on the standard config.

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBmEEBwAOgHYEEDOetOAhgB7wHUgA8sMAEsAtmwA2APl4BuOgF86IADQhSQ8TABybEXEQwOulJoB0YRoxXgItdQHMEIGvR4hDOGLSiNe3ANp0DAy8jPia4kKsRJB2QvZEYWzebGhQvEEEALoK1rEOAGIQaGI4TgBWjLbWsChWiC4hIGF4EVE4fm7ibJ5hvMqZoeEwkdH58YnsKWmdvN29HSC58kA).

Is that what you were looking for?

---

FYI, `@property` descriptors will get additional checks when https://github.com/stylelint/stylelint/issues/8212 lands.

**Comment by stof** [comment]
@jeddy3 this rule is indeed exactly what I had in mind. I missed it when searching for existing rules (I searched using `@property`, and the rule description does not mention the at-rule name at all even though it is specific to `@property`).

I'm closing this issue as it is already implemented.
